mypy/fastparse.py

```diff
@@ -294,11 +294,50 @@ def translate_expr_list(self, l: Sequence[AST]) -> List[Expression]:
             res.append(exp)
         return res
 
-    def translate_stmt_list(self, l: Sequence[AST]) -> List[Statement]:
+    def get_line(self, n: ast3.stmt) -> int:
+        if (isinstance(n, (ast3.AsyncFunctionDef, ast3.ClassDef, ast3.FunctionDef))
+                and n.decorator_list):
+            return n.decorator_list[0].lineno
+        return n.lineno
+
+    def translate_stmt_list(self,
+                            l: Sequence[ast3.stmt],
+                            module: bool = False) -> List[Statement]:
+
+        # A "# type: ignore" comment before the first statement of a module
+        # ignores the whole module:
+
+        if module and l and self.type_ignores and min(self.type_ignores) < self.get_line(l[0]):
+            self.errors.used_ignored_lines[self.errors.file].add(min(self.type_ignores))
+            b = Block(self.fix_function_overloads(self.translate_stmt_list(l)))
+            b.is_unreachable = True
+            return [b]
+
         res = []  # type: List[Statement]
-        for e in l:
+        line = 0
+
+        for i, e in enumerate(l):
+
+            if module:  # and...
+                if sys.version_info >= (3, 8):
+
+                    # In Python 3.8, a "# type: ignore" comment between statements at
+                    # the top level of a module skips checking for everything else:
```

Comment:
Most importantly, explain why: this check needs access to `end_lineno` which only exists in 3.8 and up.